server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -414,6 +423,75 @@ private void createMetadataBasedIterables(Metadata metadata) {
         routines = () -> sequentialStream(routineInfos).filter(Objects::nonNull).iterator();
     }
 
+    public Set<String> enabledRoles(Role role, Roles roles) {
```

Comment:
Please add unit test for this and the following methods.
`Roles` is an interface, so you can easily create test data, to test the methods, without the need of a full blown setup.

Reply:
Afaik, `enabled_roles` is just a view on `applicable_roles` whereas `enabled_roles` will just list the `role_name` while `applicable_roles` has more columns. Both must list all roles the use has either directly or by inheritance. [PG docs](https://www.postgresql.org/docs/current/infoschema-enabled-roles.html) defines that `these are all roles that the current user has direct or indirect, automatically inheriting membership in.`.

Reply:
@seut Then I can modify the `applicableRoles` function in `InformationSchemaIterables.java` so that it lists the required roles with full details, and make `enabled_roles` take data from that function's output itself (take only the role names).

Reply:
@clueless-skywatcher Yes, great. Thanks!